stdlib/public/RemoteInspection/TypeLowering.cpp

```diff
@@ -450,6 +450,17 @@ class TrivialEnumTypeInfo: public EnumTypeInfo {
   }
 };
 
+// Old bit hack for setting all the bits lower than
+// the highest set bit.
+static uint32_t maskForCount(uint32_t t) {
```

Comment:
Could you explain how this works @tbkka? I don't really follow 😅

Reply:
I think I get it now, you produce a mask with bits that are used by the no payload enum set and & that with the tag.

Reply:
Actually, re-reading this, it's off by one.   Anyway, here's how it should work:

If you have 6 values, the highest value is 5 (binary 0101) so you need a mask with the bottom 3 bits set (binary 0111) to preserve only the relevant count.  This old bithack just repeatedly ORs the bits into lower places to construct the appropriate mask from the maximum value.

The bug:  I derived the mask from the count rather than from the maximum value.  Fix coming soon.